rad/src/lib.rs

```diff
@@ -266,25 +285,79 @@ async fn rng_response(
     Ok(RadonReport::from_result(Ok(random_bytes), context))
 }
 
+/// Add HTTP headers from `retrieve.headers` to surf request
+fn add_http_headers(
+    mut request: RequestBuilder,
+    retrieve: &RADRetrieve,
+    context: &mut ReportContext<RadonTypes>,
+) -> Result<RequestBuilder> {
+    // Add random user agent
+    request = request.header("User-Agent", UserAgent::random());
+
+    // Only set extra headers if wip-http-post is active
+    if context.active_wips.as_ref().map(|w| w.wiphttppost()) != Some(true) {
+        return Ok(request);
+    }
+
+    // Add extra_headers from retrieve.headers
```

Comment:
TODO: also check what happens when a HTTP-GET request with headers is processed before, during, and after the hard fork.